d2ir/compile.go

```diff
@@ -255,6 +274,21 @@ func (c *compiler) compileMap(dst *Map, ast, scopeAST *d2ast.Map) {
 				ScopeMap: dst,
 				ScopeAST: scopeAST,
 			})
+		case n.Substitution != nil:
+			if !n.Substitution.Spread {
+				c.errorf(n.Import, "invalid non-spread substitution in map")
+				continue
+			}
+			// placeholder field to be resolved at the end
```

Comment:
Seems reasonable to me. Should be the same for arrays except a placeholder `*Scalar`.